pkg/sharing/replicator.go

```diff
@@ -102,31 +101,35 @@ func (s *Sharing) callChangesFeed(inst *instance.Instance) (*Changes, error) {
 	return &changes, nil
 }
 
-// Missings is a struct for the response of revs_diff
+// Missings is a struct for the response of _revs_diff
 type Missings map[string]MissingEntry
 
 // MissingEntry is a struct with the missing revisions for an id
 type MissingEntry struct {
 	Missing []string `json:"missing"`
-	// TODO possible_ancestors?
+	PAs     []string `json:"possible_ancestors"`
 }
 
-// callRevsDiff asks the other cozy to compute the revs_diff
+// callRevsDiff asks the other cozy to compute the _revs_diff
 // http://docs.couchdb.org/en/2.1.1/api/database/misc.html#db-revs-diff
 func (s *Sharing) callRevsDiff(m *Member, changes *Changes) (*Missings, error) {
 	u, err := url.Parse(m.Instance)
 	if err != nil {
 		return nil, err
 	}
-	body, err := json.Marshal(changes)
+	leafRevs := make(map[string][]string) // "doctype-docid" -> [leaf revisions]
```

Comment:
nit: looks like you know in advance the size of the map: `make(map[string][]string, len(*changes))` ?